README.md

```diff
@@ -97,44 +146,45 @@ For some packages, like `extra` we don't need its documentation or setup for pro
 
 ### Add external dependency to `default.nix`
 
-Lets say you wanted to make `extra` a dependency of your project:
+Lets say you wanted to add `extra` as dependency of your project and its not in the package set by default:
 
 ```
   myHaskellPackages = pkgs.haskell.packages.${compiler}.override {
-    overrides = se: su: {
+    overrides = hself: hsuper: {
       ### Add new dependences here
       extra =
-        se.callCabal2nix
+        hself.callCabal2nix
           "extra"
           (sources.extra)
           {};
-      "test" =
-        se.callCabal2nix
-          "test"
+      "your project name" =
+        hself.callCabal2nix
+          "your project name"
           (gitignore ./.)
           {};
     };
   };
 ```
 
-This will not only add `extra` to your project, but also build the documentation for you and add it to your local hoogle database. 
+This will not only add `extra` to your project, but also build the documentation for you. However, to get it in your local hoogle database you need to add it to your cabal file and then call `nix-shell`.
 
 ### Override dependency in `default.nix`
 
-If you want to override a dependency you add it like we did above with `extra`, but make sure the name is the same as what it is in the package set. A quick way to see what is in your modified Haskell package set is to export it and open it up with `nix repl`.
+If you want to override a dependency you add it like we did above with `extra`, just make sure the name is identical to what is in the package set. As you would expect, the name in the package set is the same as the name on Hackage. There are a few packages with multiple versions, like `zip` and `zip_1_4_1`.
```

Comment:
ditto about suggesting temporary changes to code. Although it's fine if you find it very useful.

Reply:
I think we should add the `myHaskellPackages` attribute to the final object. In other words, the end of `default.nix` would look like:

```
  {
    inherit shell;
    inherit exe;
    inherit docker;
    inherit myHaskellPackages;
  }
```

While developing I often want to know what is the version of some packages in the package set. Lets say I want to add `scotty` and I want a feature introduced in 0.11.0. I could add it to my cabal file, rebuild my shell, run `ghc-pkg dump | grep scotty` to figure out what version of scotty was in the package set or I could run `nix repl default.nix` and `> myHaskellPackages.scotty.version`. The later is a lot faster considering I would have to build and test the library in the other case.